### PR TITLE
Aggregate tokenization dataframes

### PR DESCRIPTION
**Pull request by chanteriam**
## Describe your changes
This PR creates dataframes that aggregate utterance tokenization across court cases.

## Checklist before requesting a review

- [x] The code runs successfully.

```
(supreme-court-predictions-py3.11) shaymilner@MacBook-Air-4 supreme-court-ml-predictions % make test
pytest -vs test/
========================================= test session starts ==========================================
platform darwin -- Python 3.11.1, pytest-7.2.2, pluggy-1.0.0 -- /Users/shaymilner/Library/Caches/pypoetry/virtualenvs/supreme-court-predictions-sGlGetDh-py3.11/bin/python
cachedir: .pytest_cache
rootdir: /Users/shaymilner/Documents/Harris/Spring23/Machine_Learning/final_project/supreme-court-ml-predictions
collected 1 item                                                                                       

test/test_util.py::test_get_full_pathway PASSED

========================================== 1 passed in 0.01s ===========================================
(supreme-court-predictions-py3.11) shaymilner@MacBook-Air-4 supreme-court-ml-predictions % make format
isort "supreme_court_predictions"/ test/
Fixing /Users/shaymilner/Documents/Harris/Spring23/Machine_Learning/final_project/supreme-court-ml-predictions/supreme_court_predictions/__main__.py
Fixing /Users/shaymilner/Documents/Harris/Spring23/Machine_Learning/final_project/supreme-court-ml-predictions/supreme_court_predictions/statistics/service.py
Skipped 1 files
black "supreme_court_predictions"/ test/
All done! ✨ 🍰 ✨
17 files left unchanged.
```
- [x] I ran `make lint` and have made the bulk of changes that it has requested.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,3 +35,7 @@ clean-data:
 .PHONY: describe-data
 describe-data:
 	python -m ${BASEDIR} --describe-data
+
+.PHONY: process-data
```
Comment (michplunkett) [inline Makefile:39]: Add this to the `run` command above it, por favor.

File: supreme_court_predictions/statistics/service.py [hunk 2]
```diff
@@ -3,6 +3,9 @@
 """
 
 from supreme_court_predictions.processing.datacleaner import DataCleaner
+from supreme_court_predictions.processing.token_aggregation import (
+    TokenAggregations,
+)
```
Comment (michplunkett) [inline supreme_court_predictions/statistics/service.py:8]: Long line is lonnnnnng.
image

File: supreme_court_predictions/processing/token_aggregation.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        self.advocate_tokens = None
+        self.adversary_tokens = None
+        self.judge_tokens = None
+
+        self.save_data = save_data
+
+        # Get local directory
+        self.local_path = get_full_pathway(
+            "/supreme_court_predictions/data/clean_convokit/"
+        )
```
Comment (michplunkett) [inline supreme_court_predictions/processing/token_aggregation.py:24]: I feel like I should add the `supreme_court_predictions bit` on to this function as we use it every time.

File: supreme_court_predictions/processing/token_aggregation.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+                self.all_tokens,
+                self.advocate_tokens,
+                self.adversary_tokens,
+                self.judge_tokens,
+            ]
+            outpaths = [
+                self.output_path + "/case_aggregations.p",
+                self.output_path + "/advocate_aggregations.p",
+                self.output_path + "/adversary_aggregations.p",
+                self.output_path + "/judge_aggregations.p",
```
Comment (michplunkett) [inline supreme_court_predictions/processing/token_aggregation.py:231]: What is the `.p` extension?

**Comment by michplunkett** [comment]
Looks good! If you could add the `process-data` flag to the `run` command an update the `README.md` file at the bottom with a summary of it, that'd be great.

File: supreme_court_predictions/processing/token_aggregation.py (hunk 4 above)
Comment (chanteriam) [reply supreme_court_predictions/processing/token_aggregation.py:231]: It's the extension for pickle files, easier to read and write out to than constantly using CSVs

File: supreme_court_predictions/processing/token_aggregation.py [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        """
+        Gets all the tokens for a given case and the outcome of the case.
+
+        :return A dataframe of case utterances to aggregate tokens of.
+        """
+        return self.get_case_tokens(
+            self.utterance_sides.loc[:, ["case_id", "tokens"]]
+        )
+
+    def append_side(self, utterances):
```
Comment (JJModern) [inline supreme_court_predictions/processing/token_aggregation.py:137]: looks good

**Review by JJModern — APPROVED** [review]
great!

**Pull request merged**